docs/troubleshooting.md

```diff
@@ -18,17 +18,118 @@ might overload systems if left on indefinitely.
+`SDK`. If this is non-zero that means the `BatchSpanProcessor` needs to be
+reconfigured.
+
+The `BatchSpanProcessor` has the following configuration parameters.
+
+| Configuration Parameter                        | Default | Environment Variable             |
+| ---------------------------------------------- | ------- | -------------------------------- |
+| Delay interval between two consecutive exports | 5000    | `OTEL_BSP_SCHEDULE_DELAY`        |
+| Maximum allowed time to export data            | 30000   | `OTEL_BSP_EXPORT_TIMEOUT`        |
+| Maximum queue size                             | 2048    | `OTEL_BSP_MAX_QUEUE_SIZE`        |
+| Maximum batch size                             | 512     | `OTEL_BSP_MAX_EXPORT_BATCH_SIZE` |
+
+The `BatchSpanProcessor` attempts and export when the queue contains the batch
+size or more number of spans. It will drop spans when new spans are added but
+the queue is full. Therefore there are two reasons spans are dropped: spans are
+being added faster than they can be exported, or exporting is taking so long
+the queue fills during the export.
+
+If the `count` in the log message continually equal to the maximum batch size,
+it is likely spans are being added faster than they can be exported.
+
+One way to resolve this is throw more computational and network resources at
+it. If you system has resources to spare try increasing the batch size to use
+more network bandwidth per export and increase the queue size to hold a bigger
+buffer. For example:
+
+```sh
+export OTEL_BSP_MAX_EXPORT_BATCH_SIZE=1024
+export OTEL_BSP_MAX_QUEUE_SIZE=20480
+```
+
+If the system has limited memory to spare to not increase the maximum queue
+size. If the network has not bandwidth to spare it might be better to reduce
+your export batch size. For example.
+
+```sh
+export OTEL_BSP_MAX_EXPORT_BATCH_SIZE=128
+```
+
+This should increase the export frequency and hopefully drain the queue faster.
+
+Otherwise, if the `count` value is not continually equal to the maximum batch
+size, the bottleneck is likely the export itself. The `SDK` is taking so long
+to export a batch that more spans than the queue can hold are added during this
+time. Most likely this is caused by an underlying network issue. Make sure you
+have a stable network to the target and that you have adequate bandwidth. In
+the meantime you can reduce export timeouts and increase the queue size.
+
+```sh
+# 5ms export timeout.
+export OTEL_BSP_EXPORT_TIMEOUT=5000
+export OTEL_BSP_MAX_QUEUE_SIZE=5120
+```
+
+Be sure to there are enough memory resources on your system to accommodate the
+increase in queue size.
+
+These changes will drop whole export batches that take too long. This means
+there will still be dropped data, but hopefully, if the network issues resolve,
+future exports will have a better chance of success. Failing these exports
+early will prevent future spans from being dropped.
```

Comment:
```suggestion
Make sure to allocate enough memory resources on your system to accommodate the
increase in queue size. Changes in the export configuration might result in the 
instrumentation dropping whole export batches that take too long.
```

While helpful, I'd rather eliminate the last bit, as it refers to too many hypotheticals. But happy to reinstate if you deem it important enough!